### PR TITLE
pass down stdin in 'import' check for extensions during sanity check

### PR DESCRIPTION
**Pull request by boegel**
By not passing down `stdin` into `run_cmd` in the `sanity_check` implementation for extensions, the 'import' check for R libraries is basically being skipped...

In addition, the caching mechanism for `run_cmd` should also take into account the `inp` argument.

### CODE REVIEW

**Comment by boegel** [comment]
This probably needs enhanced tests to verify the changes (and make sure the bug being fixed here isn't re-introduced).

**Comment by boegel** [comment]
Without the changes to the Seurat easyconfig in https://github.com/easybuilders/easybuild-easyconfigs/pull/4889, the sanity check fails as it should on top of this...

**Comment by boegel** [comment]
tests are enhanced to catch the bug

@easybuilders/easybuild-framework-maintainers ready for review

File: test/framework/run.py [hunk 1]
```diff
@@ -118,6 +118,22 @@ def test_run_cmd_simple(self):
         self.assertEqual(True, run_cmd("echo hello", simple=True))
         self.assertEqual(False, run_cmd("exit 1", simple=True, log_all=False, log_ok=False))
 
+    def test_run_cmd_cache(self):
```
Comment (ocaisa) [inline test/framework/run.py]: What about a test for the new `inp` argument to `run_cmd`?

Comment (boegel) [reply test/framework/run.py]: good point, added in #bc4be4154

**Pull request merged**